### DISCUSSION TITLE
Cache is growing infinitely

### DISCUSSION THREAD

**Discussion by edfr73** [osmandapp/OsmAnd#19353] [Q&A]
Hi,
I have been using Osmand every day since a few months for a motorcycle trip in South America. I noticed that the cache memory  is growing without limits, lately it occupied 36 GB of memory. Of course, it is possible to empty it manually, but why doesn't it empty automatically ? For example when data has not been accessed during some time. Or maybe there is an option I haven't seen, if so, please tell me where to find it.
Best regards.

**Comment by DmitryAlexei** [comment]
Hi
do you use online maps? If so, you can set an expiry time for each map. 
In Android, you should go to menu > maps and resources > local > map sources > select your online map > tap on a pencil at the top of the screen > set the expiry time for that map.
On the Local tab in the Maps and Resources section, you can also define what kind of data takes lots of memory and delete unnecessary items.

**Reply by edfr73** [reply]
Hi, I only use offline maps because i do not have any internet connection on the road. I imagine that the cache wich is growing is only for drawing maps more quickly. I deleted it one week ago, and now it is 13 Go. I did not see any option to remove it automaticaly.

**Reply by DmitryAlexei** [reply]
@edfr73 Please provide your profile settings. We need it to investigate your issue. Go to menu > settings > your mostly used profile > swipe to the bottom of the screen > export profile. Then simply add .txt extension to the file and post it here.

Please also indicate, do you use 3D relief?

**Reply by edfr73** [reply]
[Moto.osf.txt](https://github.com/osmandapp/OsmAnd/files/14735932/Moto.osf.txt)
Here is the profile I use more often, with it I use 3D relief.

**Reply by kosi2711** [reply]
Hi, happens for me too. If the name of the logfile "Moto.osf.txt" indicates that it's a Motorola-device, that may be a hint.

I have a Motorola G8 Power (4 GB RAM) running stock Android 10 and cache grows as soon as I activate rendering engine V2. Same has been true for stock A11 (before I downgraded).

Greets

**Reply by edfr73** [reply]
Hi, Moto.osf.txt is for Motorcycle.
I will try to use the rendering engine V1, without 3D relief.

**Comment by vshcherb** [comment]
Yes good idea to have a reasonable limit and autoclean
